[2.071.0] Document deprecated attribute as accepting AssignExp

Geod24 opened the pull request:
See D-Programming-Language/dmd#5302

FYI @Hackerpilot

schuetzm commented:
There's another location that still has `StringLiteral`:
http://dtest.thecybershadow.net/artifact/website-0350dc9b1d3a5f398b64acb6f72c8a691889bb96-227185bd781cb15fa9390f90060378dd/dlangspec.html#DeprecatedAttribute
I think it's this one:
https://github.com/D-Programming-Language/dlang.org/blob/master/spec/grammar.dd#L1640

schuetzm commented on spec/attribute.dd:
```diff
@@ -317,6 +317,17 @@ $(GNAME DeprecatedAttribute):
         oldBar();   // Deprecated: function test.oldBar is deprecated - Don't use bar
         ---------------
 
+        $(P While direct function call are not allowed, manifest constants are.
```
function call_s_

9rnsr replied:
With current master, direct function calls are *allowed*. The change is introduced while reviewing https://github.com/D-Programming-Language/dmd/pull/5302 .

I'm asking why the behavior is necessary in:
https://github.com/D-Programming-Language/dmd/pull/5302#issuecomment-182713890

Geod24 commented:
Updated following @schuetzm comments:

```diff
diff --git a/spec/grammar.dd b/spec/grammar.dd
index 12d54f0..05ca8f8 100644
--- a/spec/grammar.dd
+++ b/spec/grammar.dd
@@ -1637,7 +1637,7 @@ $(GNAME AlignAttribute):
 $(GRAMMAR
 $(GNAME DeprecatedAttribute):
     $(D deprecated)
-    $(D deprecated $(LPAREN)) $(GLINK_LEX StringLiteral) $(D $(RPAREN))
+    $(D deprecated $(LPAREN)) $(GLINK2 expression, AssignExpression) $(D $(RPAREN))
 )
```

Also reworded the part about direct function call and added an example.
